### DISCUSSION TITLE
cloudflare优质ip会不会像谷歌ip那样被耗尽

### DISCUSSION THREAD

**Discussion by ghost** [XIU2/CloudflareSpeedTest#341] [讨论]
会不会再过段时间就没有好ip了越来越卡了，就像以前goagent和xxnet那样

**Comment by XIU2** [comment]
Cloudflare CDN 的都是 Anycast IP，每个 IP 的**实际路由、归属地**都不是固定的，会时不时动态变化。
自选只是为了找到**当前时间/当前网络**下最适合自己的 IP，但这个 IP 也**仅仅适合于当前网络**下，如果其他地区、其他运营商的用户使用这个 IP 可能质量就**天差地别**了（或者说就是一般质量）。

****

当初谷歌 IP **归属地是固定**的，你找到一个速度快的 IP，那么大家用起来都很快（当然不同运营商可能有差异）。而 goagent 和 xxnet 被淘汰的主要原因是**大量谷歌 IP 被封**（剩下的 IP 地区越来越偏远，所以速度越来越慢），以及**技术没有更新迭代**导致的。

****

现在很多 CDN 都使用了 Anycast 技术，但很少像 Cloudflare 这样可以**随意指向任何一个 Anycast IP 去访问**的，一般都是局限于部分 IP 段（取决于网站掏了多少钱）。

**Comment by tec1987** [comment]
Google IP从来没有耗尽过。。。只不过技术更新后很少有人通过Google代理过墙了

**Comment by qnxsgwy** [comment]
Google IP从来没有耗尽+1，只要你有钱用GCP优质线路，网速依旧快到飞起
